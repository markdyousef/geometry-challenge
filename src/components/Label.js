import React from "react";
import styled from "styled-components";

const LABEL_COLORS = [
  { name: "a", color: "#F0CA4D" },
  { name: "b", color: "#E37B40" },
  { name: "c", color: "#ED3752" }
];

const LabelContainer = styled.div`
  ${props => `color: ${props.color}; border: 2px solid ${props.color}`};
  height: 20px;
  width: 20px;
  background: #fff;
  border-radius: 99em;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
`;

export default class extends React.PureComponent {
  render() {
    const color = LABEL_COLORS.find(d => d.name === this.props.name).color;
    return (
      <LabelContainer {...this.props} color={color}>
        <label htmlFor={this.props.name}>{this.props.name}</label>
      </LabelContainer>
    );
  }
}
