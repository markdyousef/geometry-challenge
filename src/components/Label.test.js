import React from "react";
import { shallow } from "enzyme";
import Label from "./Label";

it("renders Label", () => {
  const wrapper = shallow(<Label name="a" />);
  expect(wrapper).toMatchSnapshot();
});
