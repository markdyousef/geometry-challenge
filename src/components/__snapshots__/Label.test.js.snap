// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent
    name="a"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <label
        htmlFor="a"
>
        a
</label>,
      "color": "#F0CA4D",
      "name": "a",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "a",
        "htmlFor": "a",
      },
      "ref": null,
      "rendered": "a",
      "type": "label",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <label
          htmlFor="a"
>
          a
</label>,
        "color": "#F0CA4D",
        "name": "a",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "a",
          "htmlFor": "a",
        },
        "ref": null,
        "rendered": "a",
        "type": "label",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
